src/transformers/searchservice/createDirectAnswer.ts

```diff
@@ -1,14 +1,28 @@
-import { DirectAnswer } from '../../models/searchservice/response/DirectAnswer';
+import { FeaturedSnippetDirectAnswer, FieldValueDirectAnswer } from '../../models/searchservice/response/DirectAnswer';
+import { DirectAnswerType } from '../../models/searchservice/response/DirectAnswerType';
 import { ResultsFactory } from './ResultsFactory';
 
-export function createDirectAnswer(data: any): DirectAnswer {
-  return {
-    relatedResult: ResultsFactory.fromDirectAnswer(data.relatedItem.data),
-    verticalKey: data.relatedItem.verticalConfigId,
-    entityName: data.answer.entityName,
-    fieldName: data.answer.fieldName,
-    fieldApiName: data.answer.fieldApiName,
-    value: data.answer.value,
-    fieldType: data.answer.fieldType
-  };
+export function createDirectAnswer(data: any): FeaturedSnippetDirectAnswer | FieldValueDirectAnswer | undefined {
```

Comment:
It feels a bit funny that this function can return `undefined`. Could we throw an error instead if we don't recognize the DA type?

Reply:
sure